lib/spack/spack/util/url.py

```diff
@@ -343,7 +343,7 @@ def parse_git_url(url):
 
 
 def require_url_format(url):
-    ut = re.search(r'^(file://|http://|https://|ftp://|s3://|gs://|/)', url)
+    ut = re.search(r'^(file://|http://|https://|ftp://|s3://|gs://|git@|/)', url)
```

Comment:
Can you add a test to ensure this continues to work?  Look at https://github.com/spack/spack/pull/29592/files